Reason for Change:
        When building RDK-B, the meta-rdk dbus .bbappend applies a version
        specific patch to dbus to enable support for CCSP APIs. Currently no
        patch exists for dbus 1.16.2 (ie the OE 4.0 version), which causes
        OE 4.0 based RDK-B builds to fail during boot-up.

Source: COMCAST
License: AFL-2.1 | GPLv2+

Upstream-Status: Pending
diff -Naur dbus-1.16.2.origin/bus/activation.c dbus-1.16.2/bus/activation.c
--- dbus-1.16.2.origin/bus/activation.c	2026-07-30 12:21:35.541511591 +0000
+++ dbus-1.16.2/bus/activation.c	2026-07-30 12:23:03.414383130 +0000
@@ -185,7 +185,7 @@
 
   if (pending_activation->timeout_added)
     {
-      _dbus_loop_remove_timeout (bus_context_get_loop (pending_activation->activation->context),
+      dbus_loop_remove_timeout (bus_context_get_loop (pending_activation->activation->context),	    
                                  pending_activation->timeout);
       pending_activation->timeout_added = FALSE;
     }
@@ -1396,7 +1396,7 @@
 {
   /* FIXME use preallocated OOM messages instead of bus_wait_for_memory() */
   while (!try_send_activation_failure (pending_activation, how))
-    _dbus_wait_for_memory ();
+    dbus_wait_for_memory ();
 
   /* Destroy this pending activation */
   _dbus_hash_table_remove_string (pending_activation->activation->pending_activations,
@@ -1550,7 +1550,7 @@
 {
   BusPendingActivation *pending_activation = data;
 
-  return _dbus_loop_add_watch (
+  return dbus_loop_add_watch (
       bus_context_get_loop (pending_activation->activation->context),
       watch);
 }
@@ -1571,7 +1571,7 @@
 {
   BusPendingActivation *pending_activation = data;
 
-  _dbus_loop_toggle_watch (bus_context_get_loop (pending_activation->activation->context),
+  dbus_loop_toggle_watch (bus_context_get_loop (pending_activation->activation->context),
                            watch);
 }
 #endif
@@ -2030,7 +2030,7 @@
           return FALSE;
         }
 
-      if (!_dbus_loop_add_timeout (bus_context_get_loop (activation->context),
+      if (!dbus_loop_add_timeout (bus_context_get_loop (activation->context),
                                    pending_activation->timeout))
         {
           _dbus_verbose ("Failed to add timeout for pending activation\n");
diff -Naur dbus-1.16.2.origin/bus/bus.c dbus-1.16.2/bus/bus.c
--- dbus-1.16.2.origin/bus/bus.c	2026-07-30 12:21:35.541511591 +0000
+++ dbus-1.16.2/bus/bus.c	2026-07-30 12:23:03.414383130 +0000
@@ -121,7 +121,7 @@
 
   context = server_get_context (server);
 
-  return _dbus_loop_add_watch (context->loop, watch);
+  return dbus_loop_add_watch (context->loop, watch);
 }
 
 static void
@@ -133,7 +133,7 @@
 
   context = server_get_context (server);
 
-  _dbus_loop_remove_watch (context->loop, watch);
+  dbus_loop_remove_watch (context->loop, watch);
 }
 
 static void
@@ -145,7 +145,7 @@
 
   context = server_get_context (server);
 
-  _dbus_loop_toggle_watch (context->loop, watch);
+  dbus_loop_toggle_watch (context->loop, watch);
 }
 
 static dbus_bool_t
@@ -157,7 +157,7 @@
 
   context = server_get_context (server);
 
-  return _dbus_loop_add_timeout (context->loop, timeout);
+  return dbus_loop_add_timeout (context->loop, timeout);
 }
 
 static void
@@ -169,7 +169,7 @@
 
   context = server_get_context (server);
 
-  _dbus_loop_remove_timeout (context->loop, timeout);
+  dbus_loop_remove_timeout (context->loop, timeout);
 }
 
 static void
@@ -809,7 +809,7 @@
       goto failed;
     }
 
-  context->loop = _dbus_loop_new ();
+  context->loop = dbus_loop_new ();
   if (context->loop == NULL)
     {
       BUS_SET_OOM (error);
@@ -1277,7 +1277,7 @@
 
       if (context->loop)
         {
-          _dbus_loop_unref (context->loop);
+          dbus_loop_unref (context->loop);
           context->loop = NULL;
         }
 
diff -Naur dbus-1.16.2.origin/bus/connection.c dbus-1.16.2/bus/connection.c
--- dbus-1.16.2.origin/bus/connection.c	2026-07-30 12:21:35.545511630 +0000
+++ dbus-1.16.2/bus/connection.c	2026-07-30 12:23:03.414383130 +0000
@@ -240,7 +240,7 @@
       dbus_error_init (&error);
         
       while ((transaction = bus_transaction_new (d->connections->context)) == NULL)
-        _dbus_wait_for_memory ();
+        dbus_wait_for_memory ();
         
       if (!bus_service_remove_owner (service, connection,
                                      transaction, &error))
@@ -251,7 +251,7 @@
             {
               dbus_error_free (&error);
               bus_transaction_cancel_and_free (transaction);
-              _dbus_wait_for_memory ();
+              dbus_wait_for_memory ();
               goto retry;
             }
           else
@@ -290,7 +290,7 @@
 
   if (d->pending_unix_fds_timeout)
     {
-      _dbus_loop_remove_timeout (bus_context_get_loop (d->connections->context),
+      dbus_loop_remove_timeout (bus_context_get_loop (d->connections->context),
                                  d->pending_unix_fds_timeout);
       _dbus_timeout_unref (d->pending_unix_fds_timeout);
     }
@@ -358,7 +358,7 @@
 {
   DBusConnection *connection = data;
 
-  return _dbus_loop_add_watch (connection_get_loop (connection), watch);
+  return dbus_loop_add_watch (connection_get_loop (connection), watch);
 }
 
 static void
@@ -367,7 +367,7 @@
 {
   DBusConnection *connection = data;
   
-  _dbus_loop_remove_watch (connection_get_loop (connection), watch);
+  dbus_loop_remove_watch (connection_get_loop (connection), watch);
 }
 
 static void
@@ -376,7 +376,7 @@
 {
   DBusConnection *connection = data;
 
-  _dbus_loop_toggle_watch (connection_get_loop (connection), watch);
+  dbus_loop_toggle_watch (connection_get_loop (connection), watch);
 }
 
 static dbus_bool_t
@@ -385,7 +385,7 @@
 {
   DBusConnection *connection = data;
   
-  return _dbus_loop_add_timeout (connection_get_loop (connection), timeout);
+  return dbus_loop_add_timeout (connection_get_loop (connection), timeout);
 }
 
 static void
@@ -394,7 +394,7 @@
 {
   DBusConnection *connection = data;
   
-  _dbus_loop_remove_timeout (connection_get_loop (connection), timeout);
+  dbus_loop_remove_timeout (connection_get_loop (connection), timeout);
 }
 
 static void
@@ -406,8 +406,8 @@
   
   if (new_status != DBUS_DISPATCH_COMPLETE)
     {
-      while (!_dbus_loop_queue_dispatch (loop, connection))
-        _dbus_wait_for_memory ();
+      while (!dbus_loop_queue_dispatch (loop, connection))
+        dbus_wait_for_memory ();
     }
 }
 
@@ -487,7 +487,7 @@
   if (connections->pending_replies == NULL)
     goto failed_4;
   
-  if (!_dbus_loop_add_timeout (bus_context_get_loop (context),
+  if (!dbus_loop_add_timeout (bus_context_get_loop (context),
                                connections->expire_timeout))
     goto failed_5;
   
@@ -562,7 +562,7 @@
 
       bus_expire_list_free (connections->pending_replies);
       
-      _dbus_loop_remove_timeout (bus_context_get_loop (connections->context),
+      dbus_loop_remove_timeout (bus_context_get_loop (connections->context),
                                  connections->expire_timeout);
       
       _dbus_timeout_unref (connections->expire_timeout);
@@ -825,7 +825,7 @@
 
   if (dbus_connection_get_dispatch_status (connection) != DBUS_DISPATCH_COMPLETE)
     {
-      if (!_dbus_loop_queue_dispatch (bus_context_get_loop (connections->context), connection))
+      if (!dbus_loop_queue_dispatch (bus_context_get_loop (connections->context), connection))
         {
           bus_dispatch_remove_connection (connection);
           goto oom;
@@ -840,7 +840,7 @@
     goto oom;
 
   _dbus_timeout_disable (d->pending_unix_fds_timeout);
-  if (!_dbus_loop_add_timeout (bus_context_get_loop (connections->context),
+  if (!dbus_loop_add_timeout (bus_context_get_loop (connections->context),
                                d->pending_unix_fds_timeout))
     goto oom;
 
diff -Naur dbus-1.16.2.origin/bus/dir-watch-inotify.c dbus-1.16.2/bus/dir-watch-inotify.c
--- dbus-1.16.2.origin/bus/dir-watch-inotify.c	2026-07-30 12:21:35.545511630 +0000
+++ dbus-1.16.2/bus/dir-watch-inotify.c	2026-07-30 12:23:03.414383130 +0000
@@ -224,7 +224,7 @@
 
   if (watch != NULL)
     {
-      _dbus_loop_remove_watch (loop, watch);
+      dbus_loop_remove_watch (loop, watch);
       _dbus_watch_invalidate (watch);
       _dbus_watch_unref (watch);
       _dbus_loop_unref (loop);
@@ -266,7 +266,7 @@
       _dbus_fd_set_close_on_exec (inotify_fd);
 
       loop = bus_context_get_loop (context);
-      _dbus_loop_ref (loop);
+      dbus_loop_ref (loop);
 
       watch = _dbus_watch_new (inotify_fd, DBUS_WATCH_READABLE, TRUE,
                                _handle_inotify_watch, NULL, NULL);
@@ -278,7 +278,7 @@
           goto out;
         }
 
-      if (!_dbus_loop_add_watch (loop, watch))
+      if (!dbus_loop_add_watch (loop, watch))
         {
           bus_context_log (context, DBUS_SYSTEM_LOG_WARNING,
                            "Unable to add reload watch to main loop");
diff -Naur dbus-1.16.2.origin/bus/dispatch.c dbus-1.16.2/bus/dispatch.c
--- dbus-1.16.2.origin/bus/dispatch.c	2026-07-30 12:21:35.545511630 +0000
+++ dbus-1.16.2/bus/dispatch.c	2026-07-30 12:23:03.414383130 +0000
@@ -236,7 +236,7 @@
    * until we can.
    */
   while (!bus_connection_preallocate_oom_error (connection))
-    _dbus_wait_for_memory ();
+    dbus_wait_for_memory ();
 
   /* Ref connection in case we disconnect it at some point in here */
   dbus_connection_ref (connection);
@@ -636,7 +636,7 @@
 {
   while (dbus_connection_get_dispatch_status (connection) ==
          DBUS_DISPATCH_NEED_MEMORY)
-    _dbus_wait_for_memory ();
+    dbus_wait_for_memory ();
 
   return dbus_connection_pop_message (connection);
 }
@@ -647,7 +647,7 @@
 {
   while (dbus_connection_get_dispatch_status (connection) ==
          DBUS_DISPATCH_NEED_MEMORY)
-    _dbus_wait_for_memory ();
+    dbus_wait_for_memory ();
 
   return dbus_connection_borrow_message (connection);
 }
@@ -763,7 +763,7 @@
           if (dbus_error_has_name (&error, DBUS_ERROR_NO_MEMORY))
             {
               dbus_error_free (&error);
-              _dbus_wait_for_memory ();
+              dbus_wait_for_memory ();
               goto reget_service_info_data;
             }
           else
@@ -824,7 +824,7 @@
   _dbus_assert (s != NULL);
 
   while ((base_service = _dbus_strdup (s)) == NULL)
-    _dbus_wait_for_memory ();
+    dbus_wait_for_memory ();
 
   dbus_connection_ref (connection);
 
@@ -1057,7 +1057,7 @@
             {
               _dbus_verbose ("no memory to get service name arg from hello\n");
               dbus_error_free (&error);
-              _dbus_wait_for_memory ();
+              dbus_wait_for_memory ();
               goto retry_get_hello_name;
             }
           else
@@ -1071,7 +1071,7 @@
       _dbus_verbose ("Got hello name: %s\n", name);
 
       while (!dbus_bus_set_unique_name (connection, name))
-        _dbus_wait_for_memory ();
+        dbus_wait_for_memory ();
 
       socd.expected_kind = SERVICE_CREATED;
       socd.expected_service_name = name;
@@ -1117,7 +1117,7 @@
             {
               _dbus_verbose ("no memory to get service name arg from acquired\n");
               dbus_error_free (&error);
-              _dbus_wait_for_memory ();
+              dbus_wait_for_memory ();
               goto retry_get_acquired_name;
             }
           else
@@ -1368,7 +1368,7 @@
             {
               _dbus_verbose ("no memory to get uid by GetConnectionUnixUser\n");
               dbus_error_free (&error);
-              _dbus_wait_for_memory ();
+              dbus_wait_for_memory ();
               goto retry_get_property;
             }
           else
@@ -1516,7 +1516,7 @@
             {
               _dbus_verbose ("no memory to get pid by GetConnectionUnixProcessID\n");
               dbus_error_free (&error);
-              _dbus_wait_for_memory ();
+              dbus_wait_for_memory ();
               goto retry_get_property;
             }
           else
@@ -2110,7 +2110,7 @@
           if (dbus_error_has_name (&error, DBUS_ERROR_NO_MEMORY))
             {
               dbus_error_free (&error);
-              _dbus_wait_for_memory ();
+              dbus_wait_for_memory ();
               goto reget_service_name_arg;
             }
           else
@@ -2215,7 +2215,7 @@
           if (dbus_error_has_name (&error, DBUS_ERROR_NO_MEMORY))
             {
               dbus_error_free (&error);
-              _dbus_wait_for_memory ();
+              dbus_wait_for_memory ();
               goto reget_service_name_arg;
             }
           else
@@ -2368,7 +2368,7 @@
           if (dbus_error_has_name (&error, DBUS_ERROR_NO_MEMORY))
             {
               dbus_error_free (&error);
-              _dbus_wait_for_memory ();
+              dbus_wait_for_memory ();
               goto reget_service_name_arg;
             }
           else
@@ -4470,7 +4470,7 @@
 	    {
 	      _dbus_verbose ("no memory to list services by %s\n", method);
 	      dbus_error_free (&error);
-	      _dbus_wait_for_memory ();
+	      dbus_wait_for_memory ();
 	      goto retry_get_property;
 	    }
 	  else
diff -Naur dbus-1.16.2.origin/bus/expirelist.c dbus-1.16.2/bus/expirelist.c
--- dbus-1.16.2.origin/bus/expirelist.c	2026-07-30 12:21:35.545511630 +0000
+++ dbus-1.16.2/bus/expirelist.c	2026-07-30 12:23:03.414383130 +0000
@@ -68,7 +68,7 @@
 
   _dbus_timeout_disable (list->timeout);
 
-  if (!_dbus_loop_add_timeout (list->loop, list->timeout))
+  if (!dbus_loop_add_timeout (list->loop, list->timeout))
     goto failed;
 
   return list;
@@ -87,7 +87,7 @@
 {
   _dbus_assert (list->items == NULL);
 
-  _dbus_loop_remove_timeout (list->loop, list->timeout);
+  dbus_loop_remove_timeout (list->loop, list->timeout);
 
   _dbus_timeout_unref (list->timeout);
 
@@ -161,7 +161,7 @@
            */
           if (!(* list->expire_func) (list, link, list->data))
             {
-              next_interval = _dbus_get_oom_wait ();
+              next_interval = dbus_get_oom_wait ();
               break;
             }
         }
diff -Naur dbus-1.16.2.origin/bus/main.c dbus-1.16.2/bus/main.c
--- dbus-1.16.2.origin/bus/main.c	2026-07-30 12:21:35.545511630 +0000
+++ dbus-1.16.2/bus/main.c	2026-07-30 12:23:03.414383130 +0000
@@ -280,7 +280,7 @@
   char action = '\0';
 
   while (!_dbus_string_init (&str))
-    _dbus_wait_for_memory ();
+    dbus_wait_for_memory ();
 
   if ((reload_pipe[RELOAD_READ_END].fd > 0) &&
       _dbus_read_socket (reload_pipe[RELOAD_READ_END], &str, 1) != 1)
@@ -330,7 +330,7 @@
         if (loop != NULL)
           {
             _dbus_daemon_report_stopping ();
-            _dbus_loop_quit (loop);
+            dbus_loop_quit (loop);
           }
       }
       break;
@@ -371,7 +371,7 @@
       exit (1);
     }
 
-  if (!_dbus_loop_add_watch (loop, watch))
+  if (!dbus_loop_add_watch (loop, watch))
     {
       _dbus_warn ("Unable to add reload watch to main loop: %s",
 		  error.message);
@@ -384,7 +384,7 @@
 static void
 close_reload_pipe (DBusWatch **watch)
 {
-    _dbus_loop_remove_watch (bus_context_get_loop (context), *watch);
+    dbus_loop_remove_watch (bus_context_get_loop (context), *watch);
     _dbus_watch_invalidate (*watch);
     _dbus_watch_unref (*watch);
     *watch = NULL;
@@ -746,7 +746,7 @@
 
   _dbus_verbose ("We are on D-Bus...\n");
   _dbus_daemon_report_ready ();
-  _dbus_loop_run (bus_context_get_loop (context));
+  dbus_loop_run (bus_context_get_loop (context));
 
   bus_context_shutdown (context);
   bus_context_unref (context);
diff -Naur dbus-1.16.2.origin/bus/utils.c dbus-1.16.2/bus/utils.c
--- dbus-1.16.2.origin/bus/utils.c	2026-07-30 12:21:35.549511670 +0000
+++ dbus-1.16.2/bus/utils.c	2026-07-30 12:23:03.414383130 +0000
@@ -44,7 +44,7 @@
   DBusDispatchStatus status;
 
   while ((status = dbus_connection_dispatch (connection)) == DBUS_DISPATCH_NEED_MEMORY)
-    _dbus_wait_for_memory ();
+    dbus_wait_for_memory ();
   
   return status == DBUS_DISPATCH_DATA_REMAINS;
 }
diff -Naur dbus-1.16.2.origin/dbus/CMakeLists.txt dbus-1.16.2/dbus/CMakeLists.txt
--- dbus-1.16.2.origin/dbus/CMakeLists.txt	2026-07-30 12:21:35.553511710 +0000
+++ dbus-1.16.2/dbus/CMakeLists.txt	2026-07-30 12:23:03.414383130 +0000
@@ -6,6 +6,7 @@
     dbus.h
     dbus-address.h
     dbus-bus.h
+    dbus-ccsp-apis.h
     dbus-connection.h
     dbus-errors.h
     dbus-macros.h
@@ -31,6 +32,7 @@
     dbus-address.c
     dbus-auth.c
     dbus-bus.c
+    dbus-ccsp-apis.c
     dbus-connection.c
     dbus-credentials.c
     dbus-errors.c
@@ -55,7 +57,6 @@
     dbus-threads.c
     dbus-transport.c
     dbus-transport-socket.c
-    dbus-watch.c
 )
 
 
@@ -96,7 +97,6 @@
     dbus-transport.h
     dbus-transport-protected.h
     dbus-uuidgen.h
-    dbus-watch.h
     ${PROJECT_BINARY_DIR}/config.h
 )
 if(UNIX)
@@ -120,6 +120,10 @@
     dbus-hash.c
     dbus-internals.c
     dbus-list.c
+    dbus-mainloop.c
+    dbus-pollable-set.c
+    dbus-pollable-set-poll.c
+    dbus-watch.c
     dbus-marshal-basic.c
     dbus-memory.c
     dbus-mempool.c
@@ -134,6 +138,9 @@
     dbus-hash.h
     dbus-internals.h
     dbus-list.h
+    dbus-watch.h
+    dbus-pollable-set.h
+    dbus-mainloop.h
     dbus-macros-internal.h
     dbus-marshal-basic.h
     dbus-mempool.h
@@ -226,8 +233,8 @@
 endif()
 
 if(DBUS_HAVE_LINUX_EPOLL)
-    set(DBUS_UTIL_SOURCES
-        ${DBUS_UTIL_SOURCES}
+    set(DBUS_SHARED_SOURCES
+        ${DBUS_SHARED_SOURCES}
         dbus-pollable-set-epoll.c
     )
 endif()
diff -Naur dbus-1.16.2.origin/dbus/dbus-ccsp-apis.c dbus-1.16.2/dbus/dbus-ccsp-apis.c
--- dbus-1.16.2.origin/dbus/dbus-ccsp-apis.c	1970-01-01 00:00:00.000000000 +0000
+++ dbus-1.16.2/dbus/dbus-ccsp-apis.c	2026-07-30 12:23:57.246917128 +0000
@@ -0,0 +1,147 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/* dbus-ccsp-apis.c
+ *  *
+ *   * Copyright (C) 2003  Red Hat, Inc.
+ *    *
+ *     * Licensed under the Academic Free License version 2.1
+ *      *
+ *       * This program is free software; you can redistribute it and/or modify
+ *        * it under the terms of the GNU General Public License as published by
+ *         * the Free Software Foundation; either version 2 of the License, or
+ *          * (at your option) any later version.
+ *           *
+ *            * This program is distributed in the hope that it will be useful,
+ *             * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *              * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *               * GNU General Public License for more details.
+ *                *
+ *                 * You should have received a copy of the GNU General Public License
+ *                  * along with this program; if not, write to the Free Software
+ *                   * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *                    *
+ *                     */
+
+#include <config.h>
+#include "dbus-ccsp-apis.h"
+#include "dbus-internals.h"
+#include "dbus-watch.h"
+#ifdef DBUS_UNIX
+#include <dbus/dbus-sysdeps-unix.h>
+#endif
+
+#include "dbus-mainloop.h"
+#include "dbus-connection-internal.h"
+#include <unistd.h>
+
+/* Handle the loop wakeup watch */
+dbus_bool_t wake_handler(DBusWatch *watch, unsigned int flags, void *data)
+{
+	    char ch;
+	        int fd = (int)data;
+		    /* Read and discard */
+		    while(read(fd, &ch, 1) == 1);
+}
+
+/* We only need 1 wake watch per mainloop process */
+DBusWatch *wake_watch = NULL;
+
+/* From dbus-mainloop */
+DBusLoop* dbus_loop_new(void)
+{
+	    return _dbus_loop_new();
+}
+
+DBusLoop* dbus_loop_ref(DBusLoop *loop)
+{
+	    return _dbus_loop_ref(loop);
+}
+
+void dbus_loop_unref(DBusLoop *loop) { _dbus_loop_unref(loop); }
+
+dbus_bool_t dbus_loop_add_watch(DBusLoop *loop, DBusWatch *watch)
+{
+	    return _dbus_loop_add_watch(loop, watch);
+}
+
+void dbus_loop_remove_watch(DBusLoop *loop, DBusWatch *watch)
+{
+	    _dbus_loop_remove_watch(loop, watch);
+}
+
+void dbus_loop_toggle_watch(DBusLoop *loop, DBusWatch *watch)
+{
+	    _dbus_loop_toggle_watch(loop, watch);
+}
+
+dbus_bool_t dbus_loop_add_timeout(DBusLoop *loop, DBusTimeout *timeout)
+{
+	    return _dbus_loop_add_timeout(loop, timeout);
+}
+
+void dbus_loop_remove_timeout(DBusLoop *loop, DBusTimeout *timeout)
+{
+	     _dbus_loop_remove_timeout(loop, timeout);
+}
+
+dbus_bool_t dbus_loop_queue_dispatch(DBusLoop *loop, DBusConnection *connection)
+{
+	    return _dbus_loop_queue_dispatch(loop, connection);
+}
+
+void dbus_loop_run(DBusLoop *loop)
+{
+	     _dbus_loop_run(loop);
+}
+
+void dbus_loop_quit(DBusLoop *loop)
+{
+	     _dbus_loop_quit(loop);
+}
+
+dbus_bool_t dbus_loop_iterate(DBusLoop *loop, dbus_bool_t block)
+{
+	    return _dbus_loop_iterate(loop, block);
+}
+
+dbus_bool_t dbus_loop_dispatch(DBusLoop *loop)
+{
+	    return _dbus_loop_dispatch(loop);
+}
+
+void dbus_wait_for_memory(void)
+{
+	    _dbus_wait_for_memory();
+}
+
+/* From dbus-connection */
+void dbus_connection_lock(DBusConnection *connection)
+{
+	    _dbus_connection_lock(connection);
+}
+
+void dbus_connection_unlock(DBusConnection *connection)
+{
+	    _dbus_connection_unlock(connection);
+}
+
+int dbus_get_oom_wait (void)
+{
+	     return _dbus_get_oom_wait();
+}
+
+dbus_bool_t dbus_loop_add_wake(DBusLoop *loop, int fd)
+{
+	     wake_watch = _dbus_watch_new(fd, DBUS_WATCH_READABLE, TRUE, wake_handler, (void *)fd, NULL);
+	          return dbus_loop_add_watch(loop, wake_watch);
+}
+
+void dbus_loop_remove_wake(DBusLoop *loop)
+{
+	     if (wake_watch)
+		          {
+				           dbus_loop_remove_watch(loop, wake_watch);
+					            _dbus_watch_unref(wake_watch);
+						             wake_watch = NULL;
+							          }
+}
+
diff -Naur dbus-1.16.2.origin/dbus/dbus-ccsp-apis.h dbus-1.16.2/dbus/dbus-ccsp-apis.h
--- dbus-1.16.2.origin/dbus/dbus-ccsp-apis.h	1970-01-01 00:00:00.000000000 +0000
+++ dbus-1.16.2/dbus/dbus-ccsp-apis.h	2026-07-30 12:23:03.414383130 +0000
@@ -0,0 +1,112 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/* dbus-ccsp-apis.h
+ *
+ * Copyright (C) 2003  Red Hat, Inc.
+ *
+ * Licensed under the Academic Free License version 2.1
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+#if !defined (DBUS_INSIDE_DBUS_H) && !defined (DBUS_COMPILATION)
+#error "Only <dbus/dbus.h> can be included directly, this file may disappear or change contents."
+#endif
+
+#ifndef DBUS_CCSP_APIS_H
+#define DBUS_CCSP_APIS_H
+
+#include <dbus/dbus.h>
+#include <stdarg.h>
+
+typedef struct DBusLoop DBusLoop;
+
+DBUS_BEGIN_DECLS
+
+/**
+ * @addtogroup DBusCcspApis
+ * @{
+ */
+
+/* From dbus-mainloop */
+DBUS_EXPORT
+DBusLoop*   dbus_loop_new            (void);
+
+DBUS_EXPORT
+DBusLoop*   dbus_loop_ref            (DBusLoop            *loop);
+
+DBUS_EXPORT
+void        dbus_loop_unref          (DBusLoop            *loop);
+
+DBUS_EXPORT
+dbus_bool_t dbus_loop_add_watch      (DBusLoop            *loop,
+                                      DBusWatch           *watch);
+
+DBUS_EXPORT
+void        dbus_loop_remove_watch   (DBusLoop            *loop,
+                                      DBusWatch           *watch);
+DBUS_EXPORT
+void        dbus_loop_toggle_watch   (DBusLoop            *loop,
+                                      DBusWatch           *watch);
+DBUS_EXPORT
+dbus_bool_t dbus_loop_add_timeout    (DBusLoop            *loop,
+                                      DBusTimeout         *timeout);
+
+DBUS_EXPORT
+void        dbus_loop_remove_timeout (DBusLoop            *loop,
+                                      DBusTimeout         *timeout);
+
+DBUS_EXPORT
+dbus_bool_t dbus_loop_queue_dispatch (DBusLoop            *loop,
+                                      DBusConnection      *connection);
+
+DBUS_EXPORT
+void        dbus_loop_run            (DBusLoop            *loop);
+
+DBUS_EXPORT
+void        dbus_loop_quit           (DBusLoop            *loop);
+
+DBUS_EXPORT
+dbus_bool_t dbus_loop_iterate        (DBusLoop            *loop,
+                                      dbus_bool_t          block);
+
+DBUS_EXPORT
+dbus_bool_t dbus_loop_dispatch       (DBusLoop            *loop);
+
+DBUS_EXPORT
+void dbus_wait_for_memory(void);
+
+DBUS_EXPORT
+int dbus_get_oom_wait(void);
+
+/* From dbus-connection */
+DBUS_EXPORT
+void dbus_connection_lock(DBusConnection *connection);
+
+DBUS_EXPORT
+void dbus_connection_unlock(DBusConnection *connection);
+
+DBUS_EXPORT
+dbus_bool_t dbus_loop_add_wake       (DBusLoop            *loop,
+                                      int                 fd);
+
+DBUS_EXPORT
+void dbus_loop_remove_wake           (DBusLoop            *loop);
+
+/** @} */
+
+DBUS_END_DECLS
+
+#endif /* DBUS_CCSP_APIS_H */
+
diff -Naur dbus-1.16.2.origin/dbus/dbus.h dbus-1.16.2/dbus/dbus.h
--- dbus-1.16.2.origin/dbus/dbus.h	2026-07-30 12:21:35.573511908 +0000
+++ dbus-1.16.2/dbus/dbus.h	2026-07-30 12:23:03.414383130 +0000
@@ -31,6 +31,7 @@
 #include <dbus/dbus-arch-deps.h>
 #include <dbus/dbus-address.h>
 #include <dbus/dbus-bus.h>
+#include <dbus/dbus-ccsp-apis.h>
 #include <dbus/dbus-connection.h>
 #include <dbus/dbus-errors.h>
 #include <dbus/dbus-macros.h>
diff -Naur dbus-1.16.2.origin/dbus/meson.build dbus-1.16.2/dbus/meson.build
--- dbus-1.16.2.origin/dbus/meson.build	2026-07-30 12:21:35.573511908 +0000
+++ dbus-1.16.2/dbus/meson.build	2026-07-30 12:23:57.246917128 +0000
@@ -78,7 +78,18 @@
     'dbus-string.c',
     'dbus-sysdeps.c',
 ]
-
+dbus_lib_sources += [
+    'dbus-ccsp-apis.c',
+    'dbus-asv-util.c',
+    'dbus-mainloop.c',
+    'dbus-message-util.c',
+    'dbus-pollable-set-epoll.c',
+    'dbus-pollable-set-poll.c',
+    'dbus-pollable-set.c',
+    'dbus-shell.c',
+    'dbus-string-util.c',
+    'dbus-sysdeps-util.c',
+]
 if intrusive_tests
     dbus_shared_sources += 'dbus-test-tap.c'
 endif
@@ -168,9 +179,13 @@
 
 
 if use_linux_epoll
+    dbus_shared_sources += 'dbus-pollable-set-epoll.c'
+endif
+if use_linux_epoll
     dbus_util_sources += 'dbus-pollable-set-epoll.c'
 endif
 
+
 version_script = configure_file(
     input: 'Version.in',
     output: 'version_script',
